src/utils/LoggingUtil.ts

```diff
@@ -0,0 +1,32 @@
+import { TEST_STATUS, TEST_RESULT } from "../assets/Enums";
+import { Defect, TestType } from "../models/ITestResult";
+import { ITestResultPayload } from "../models/ITestResultPayload";
+
+export class LoggingUtil {
+    private static readonly reportingDeficiencyRef = ["8.1.i", "8.1.j.i", "8.1.j.ii", "8.1", "8.2"];
+    /**
+     * Function to log certain defects and advisory silently
+     */
+    public static logDefectsReporting(testResult: ITestResultPayload): void {
+
+        if (testResult.testStatus === TEST_STATUS.CANCELLED) { return; }
+
+        testResult.testTypes.forEach((testType: TestType) => {
+            if (testType.testResult !== TEST_RESULT.ABANDONED) {
+                testType.defects.forEach((defect: Defect) => {
+                    if (LoggingUtil.reportingDeficiencyRef.includes(defect.deficiencyRef)) {
+                        const logObject = {vin: testResult.vin,
+                            vrm: testResult.vrm,
+                            additionalNotesRecorded: testType.additionalNotesRecorded,
+                            deficiencyRef: defect.deficiencyRef,
+                            deficiencyCategory: defect.deficiencyCategory,
+                            defectNotes: defect?.additionalInformation.notes,
+                            ...defect?.additionalInformation?.location
+                        };
+                        console.info("Defects reporting: ", logObject);
+                    }
+                });
+            }
+        });
+    }
```

Comment:
This code just constructs the object by accessing the fields safely and displays a console log. I had the try catch initially but it seemed unnecessary mostly because all the checker were done before this steps and the test-result was saved in the database. If you think a try-catch will improve the safety, I can add it back